src/strings/nfa-strings.cc

```diff
@@ -147,6 +147,17 @@ void ShortestWordsMap::update_current_words(LengthWordsPair& act, const LengthWo
     act.first = dst.first + 1;
 }
 
+std::set<Mata::Symbol> Mata::Strings::get_one_symbol_words(const Nfa::Nfa& nfa) {
```

Comment:
rewrite with the new david's iterator? Or not?

Reply:
I am not sure if it makes sense.

Reply:
I agree that using moves is not efficient here. You do not want to have to iterate over each target for the same symbol separately. You want to check the emptiness of the intersection for all targets together.